feat(): add custom ns labels/annotations to appns

gourishkb opened the pull request:
user defined CR labels/annotations config is fetched from cluster CR and configmap "namespace-config-labels" is created
this configmap is used to apply labels/annotations to any slice appns

When a slice appns undergoes unbinding, the custom labels/annotations are also removed

vendor/ changes will be updated with apis repo tag when changes are approved

<!--
Type of change: Title of the PR should clearly mention which type of PR is this, you can select any of the below-mentioned types:

- docs() - The PR contains Documentation ONLY changes. 
- feat() - The PR contains new feature/enhancements.
- fix() - The PR contains a bug fix.

Example Title: 
feat(): New field addition for Cluster CRD 
-->

# Description
<!--
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
-->

Fixes # <!-- Mention any issues which might be fixed on this PR merge. -->

## How Has This Been Tested?
<!--
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

test-cases
- [ ] Test case A
- [ ] Test case B
-->

## Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have ran `go fmt`
* [ ] I have updated the helm chart as required by this PR.
* [ ] I have performed a self-review of my own code.
* [ ] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit test cases.
* [ ] I have verified the E2E test cases with new code changes.
* [ ] I have added all the required E2E test cases.

## Does this PR introduce a breaking change?
<!--
If NO, leave the release-note block blank.
If YES, a release note is required:
Enter your extended release note in the block below. If the PR requires additional manual action from users switching to the new version, include the string "action-required".
-->

```release-note

```

gourishkb committed:
feat(): add custom ns labels/annotations to appns

user defined CR labels/annotations config is fetched from cluster CR and
configmap "namespace-config-labels" is created
this configmap is used to apply labels/annotations to any slice appns

When a slice appns undergoes unbinding, the custom labels/annotations
are also removed

vendor/ changes will be updated with apis repo tag when changes are
approved

Signed-off-by: gourishkb <104021126+gourishkb@users.noreply.github.com>

gourishkb committed:
Revert "feat(): add custom ns labels/annotations to appns"

This reverts commit 254111abd197d9a3d29ec08ba35c105d50f958e4.

gourishkb committed:
fix(): add changed manifest files

Signed-off-by: gourishkb <104021126+gourishkb@users.noreply.github.com>

Rahul-D78 commented:
@gourishkb why are we creating the "namespace-config-labels" configmap ?? can't we fetch the namespace labels and annotation from the CR directly from the namespace reconciler ??

gourishkb committed:
fix(): address review comments

use cluster CR to fetch namespace config instead of configMap create and
fetch for namespace labels/annotations

Signed-off-by: gourishkb <104021126+gourishkb@users.noreply.github.com>

gourishkb committed:
fix(): address review comments

use cluster CR to fetch namespace config instead of configMap create and
fetch for namespace labels/annotations

Signed-off-by: gourishkb <104021126+gourishkb@users.noreply.github.com>

gourishkb commented:
addressed review comments from @Rahul-D78 , testing pending for the changes. moving PR to draft until testing is done

gourishkb committed:
fix(): get controller cluster CR

use hubClient interface to fetch the controller cluster CR
namespaceConfig

Signed-off-by: gourishkb <104021126+gourishkb@users.noreply.github.com>

gourishkb commented:
Tested multiple scenarios with adding, removing labels/annotations and namespaces. works as expected. Opening the PR for review

gourishkb committed:
fix(): integrate with projectns labels/annotations

Signed-off-by: gourishkb <104021126+gourishkb@users.noreply.github.com>
